### PR TITLE
Introduce lineId for log entries

### PR DESCRIPTION
**Pull request by algairim**
Signed-off-by: Mykola Mandra <mykola.mandra@cloudsoft.io>

### CODE REVIEW

File: core/src/main/java/org/apache/brooklyn/util/core/logbook/file/FileLogStore.java [hunk 1]
```diff
@@ -69,23 +70,28 @@
...
+    private final DateFormat dateFormat;
+
 
     @VisibleForTesting
     public FileLogStore() {
         this.path = null;
         this.filePath = "";
         this.logLinePattern = LOGBOOK_LOG_STORE_REGEX.getDefaultValue();
-        this.logDateFormat = LOGBOOK_LOG_STORE_DATEFORMAT.getDefaultValue();
+        dateFormat = new SimpleDateFormat(LOGBOOK_LOG_STORE_DATEFORMAT.getDefaultValue());
```
Comment (jcabrerizo) [inline core/src/main/java/org/apache/brooklyn/util/core/logbook/file/FileLogStore.java:86]: I'd add `this.` to dateFormat to keep it consistent

**Review by jcabrerizo — COMMENTED** [review]
LGTM @algairim

File: core/src/main/java/org/apache/brooklyn/util/core/logbook/file/FileLogStore.java (hunk 1 above)
Comment (ahgittin) [reply core/src/main/java/org/apache/brooklyn/util/core/logbook/file/FileLogStore.java:86]: personally i don't like the `this.` (except where it is needed)

File: core/src/test/java/org/apache/brooklyn/util/core/logbook/file/FileLogStoreTest.java [hunk 2]
```diff
@@ -48,64 +49,74 @@
             "\tat org.apache.felix.scr.impl.ComponentRegistry.checkComponentName(ComponentRegistry.java:240) ~[?:?]\n" +
             "\tat org.apache.felix.scr.impl.BundleComponentActivator.validateAndRegister(BundleComponentActivator.java:443) ~[?:?]";
 
+    private final AtomicInteger lineCount = new AtomicInteger();
+
     @BeforeTest
     public void setUp() {
-        TimeZone.setDefault(TimeZone.getTimeZone("GMT"));
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
```
Comment (ahgittin) [inline core/src/test/java/org/apache/brooklyn/util/core/logbook/file/FileLogStoreTest.java:56]: i don't like having to set this in a test but don't have a better idea so go for it

**Comment by ahgittin** [comment]
LGTM

**Pull request merged**